tests/media/server/ipc/sessionManagementServer/SessionManagementServerTestsFixture.cpp

```diff
@@ -182,12 +194,107 @@ void SessionManagementServerTests::serverWillSetLogLevels()
 
 void SessionManagementServerTests::sendServerInitialize()
 {
-    EXPECT_TRUE(m_sut->initialize(socketName, socketPermissions));
+    EXPECT_CALL(*m_linuxWrapperMock, chmod(testing::StrEq(socketName), socketPermissions)).WillOnce(Return(0));
+    EXPECT_TRUE(m_sut->initialize(socketName, socketPermissions, socketOwner, socketGroup));
 }
 
 void SessionManagementServerTests::sendServerInitializeAndExpectFailure()
 {
-    EXPECT_FALSE(m_sut->initialize(socketName, socketPermissions));
+    EXPECT_FALSE(m_sut->initialize(socketName, socketPermissions, socketOwner, socketGroup));
+}
+
+void SessionManagementServerTests::testSocketOwnershipValidOwnerAndGroup()
+{
+    serverWillInitialize();
+    EXPECT_CALL(*m_linuxWrapperMock, chmod(testing::StrEq(socketName), socketPermissions)).WillOnce(Return(0));
+
+    struct passwd passwordStruct
+    {
+    };
+    passwordStruct.pw_uid = kTestOwnerId;
+    EXPECT_CALL(*m_linuxWrapperMock, getpwnam_r(testing::StrEq(kTestOwner), _, _, _, _))
+        .WillOnce(testing::DoAll(testing::SetArgPointee<4>(&passwordStruct), Return(0)));
+
+    struct group groupStruct
+    {
+    };
+    groupStruct.gr_gid = kTestGroupId;
+    EXPECT_CALL(*m_linuxWrapperMock, getgrnam_r(testing::StrEq(kTestGroup), _, _, _, _))
+        .WillOnce(testing::DoAll(testing::SetArgPointee<4>(&groupStruct), Return(0)));
+
+    EXPECT_CALL(*m_linuxWrapperMock, chown(testing::StrEq(socketName), kTestOwnerId, kTestGroupId)).WillOnce(Return(0));
+
+    EXPECT_TRUE(m_sut->initialize(socketName, socketPermissions, kTestOwner, kTestGroup));
```

Comment:
Instead of having invocation in this function, can you just call sendServerInitialize from the test cases themselves in SessionManagementServerTests.cpp.